src/retrace/retrace.py

```diff
@@ -1310,6 +1310,51 @@ def package_name(self, debug=False):
     def needs_arch(self):
         return self._arch is None
 
+class KernelVMcore:
+    def __init__(self, vmcore_path):
+        self._vmcore_path = vmcore_path
+        self._is_flattened_format = None
+
+    def is_flattened_format(self):
+        """Returns True if vmcore is in makedumpfile flattened format"""
+        if self._is_flattened_format is not None:
+            return self._is_flattened_format
+        try:
+            with open(self._vmcore_path) as fd:
+                fd.seek(0)
+                # Read 16 bytes (SIG_LEN_MDF from crash-utility makedumpfile.h)
+                b = os.read(fd.fileno(), 16)
```

Comment:
Oops, I missed this before: You can simply call `fd.read(16)` to read from the file. You also need to open the file in `rb` mode above, so that it returns bytes and not a string.

Reply:
nice catch - fixing...

Reply:
resolved in 05cbb12